### PR TITLE
Rename migration-engine parts to schema-engine parts

### PR DESCRIPTION
**Pull request by pimeys**
The last step in merging the intro and migrations.

- ~Renaming the `migration-engine` binary to `schema-engine`.~
	- Keeping the `migration-engine` binary name as before, rename happens on Prisma 5. 
- `MigrationConnector` is now `SchemaConnector`.
- `SqlMigrationConnector` is now `SqlSchemaConnector`.
- `MongoDbMigrationConnector` is now `MongoDbSchemaConnector`.
- `introspection-engine-tests` is now `sql-introspection-tests`
- `migration-engine-tests` is now `sql-migration-tests`.
- `migration_api` is now `schema_api`.

~Requires attention from @jkomyno and/or @Druue when merging to fix the client stuff.~

### CODE REVIEW

File: libs/test-cli/Cargo.toml [hunk 1]
```diff
@@ -9,8 +9,8 @@ colored = "2"
 structopt = "0.3.8"
 enumflags2 = "0.7"
 dmmf = { path = "../../query-engine/dmmf" }
-migration-core = { path = "../../migration-engine/core" }
```
Comment (Druue) [inline libs/test-cli/Cargo.toml:12]: For context, as I'm not super aware of what this looked like prior, whilst we still had an official introspection engine: Did we have an introspection-core, and was that then merged into migration-core too? Or does the new schema-core currently only handle migrations

File: libs/user-facing-errors/src/schema_engine.rs [hunk 2]
```diff
@@ -267,6 +267,20 @@ pub struct ForeignKeyCreationNotAllowed;
 )]
 pub struct DirectDdlNotAllowed;
 
+#[derive(Debug, SimpleUserFacingError)]
```
Comment (Druue) [inline libs/user-facing-errors/src/schema_engine.rs:270]: These were brought over from the deleted `introspection_engine.rs` right? We're missing code `4000`?
> Introspection operation failed to produce a schema file

File: query-engine/connector-test-kit-rs/qe-setup/src/cockroachdb.rs [hunk 3]
```diff
@@ -1,6 +1,6 @@
-use migration_core::migration_connector::{ConnectorError, ConnectorResult};
 use once_cell::sync::OnceCell;
 use quaint::{prelude::*, single::Quaint};
+use schema_core::schema_connector::{ConnectorError, ConnectorResult};
```
Comment (Druue) [inline query-engine/connector-test-kit-rs/qe-setup/src/cockroachdb.rs:3]: Is this something that we want to update the naming convention of, wrt `migration_connector` -> `schema_connector` or are these used for introspection too?

File: schema-engine/sql-migration-tests/tests/initialization/mod.rs [hunk 4]
```diff
@@ -53,10 +53,10 @@ fn connecting_to_a_postgres_database_with_missing_schema_creates_it(api: TestApi
                 "#
         );
 
-        let me = migration_api(Some(schema.clone()), None).unwrap();
+        let me = schema_api(Some(schema.clone()), None).unwrap();
```
Comment (Druue) [inline schema-engine/sql-migration-tests/tests/initialization/mod.rs:56]: Is `me` supposed to stand for "migration engine"? This reads a little unclearly to me

Comment (Druue) [reply schema-engine/sql-migration-tests/tests/initialization/mod.rs:56]: We seem to use this in other areas too

File: schema-engine/sql-migration-tests/tests/migrations/cockroachdb.rs [hunk 5]
```diff
@@ -27,8 +27,8 @@ fn db_push_on_cockroach_db_with_postgres_provider_works(api: TestApi) {
         api.connection_string()
     );
 
-    let connector = migration_core::migration_api(Some(schema.clone()), None).unwrap();
-    let output = tok(connector.schema_push(migration_core::json_rpc::types::SchemaPushInput {
+    let connector = schema_core::schema_api(Some(schema.clone()), None).unwrap();
+    let output = tok(connector.schema_push(schema_core::json_rpc::types::SchemaPushInput {
```
Comment (Druue) [inline schema-engine/sql-migration-tests/tests/migrations/cockroachdb.rs:31]: I tried to check out what `tok` referred to and only managed to find the following:
```rs
run_with_thread_local_runtime
```
Does "tok" mean something specific here? It's unclear to me

File: schema-engine/sql-migration-tests/tests/migrations/db_execute.rs [hunk 6]
```diff
@@ -32,7 +32,7 @@ fn db_execute_happy_path_with_literal_url() {
     "#;
 
     // Execute the command.
-    let generic_api = migration_core::migration_api(None, None).unwrap();
+    let generic_api = schema_core::schema_api(None, None).unwrap();
```
Comment (Druue) [inline schema-engine/sql-migration-tests/tests/migrations/db_execute.rs:35]: is `generic_api` supposed to reference the set of functionality across all connectors?

**Review by Druue — COMMENTED** [review]
LGTM, just have a couple of questions & comments 😊

File: .github/workflows/schema-engine.yml [hunk 7]
```diff
@@ -240,6 +240,6 @@ jobs:
           sudo sc start MySQL
 
       - name: Run tests
-        run: cargo test -p migration-engine-tests
+        run: cargo test -p sql-migration-tests
```
[the remark this answers is not in the document]
Comment (pimeys) [reply .github/workflows/schema-engine.yml:243]: We do not really test the engine, we test the sql connector here.

File: libs/test-cli/Cargo.toml (hunk 1 above)
Comment (pimeys) [reply libs/test-cli/Cargo.toml:12]: We had introspection-core before, the schema-core handles both.

File: libs/user-facing-errors/src/schema_engine.rs (hunk 2 above)
Comment (pimeys) [reply libs/user-facing-errors/src/schema_engine.rs:270]: The code 4000 is not used anywhere in the codebase.

File: query-engine/connector-test-kit-rs/qe-setup/src/cockroachdb.rs (hunk 3 above)
Comment (pimeys) [reply query-engine/connector-test-kit-rs/qe-setup/src/cockroachdb.rs:3]: The connector is also used for introspection.

File: schema-engine/sql-migration-tests/tests/migrations/cockroachdb.rs (hunk 5 above)
Comment (pimeys) [reply schema-engine/sql-migration-tests/tests/migrations/cockroachdb.rs:31]: It runs async code in a sync context. `tok` meaning `tokio::spawn(future).join()`.

File: .github/workflows/schema-engine.yml (hunk 7 above)
[the remark this answers is not in the document]
Comment (pimeys) [reply .github/workflows/schema-engine.yml:243]: Or, let me rephrase, we have two rounds of tests: one for migrations and one for introspection. These things will probably never merge, we want them as separate tests.

So, what they test is either SQL migrations or SQL introspection. There's no migration-engine or introspection-engine in the codebase, so the current names are not good.

File: schema-engine/sql-migration-tests/tests/initialization/mod.rs (hunk 4 above)
Comment (pimeys) [reply schema-engine/sql-migration-tests/tests/initialization/mod.rs:56]: Should maybe rename to `engine`... It's test code so I'm not too eager to do that work.

File: schema-engine/sql-migration-tests/tests/migrations/db_execute.rs (hunk 6 above)
Comment (pimeys) [reply schema-engine/sql-migration-tests/tests/migrations/db_execute.rs:35]: Yep. So that's the JSON RPC API that can do introspection, migrations, diffing etc. for all the connectors.

File: libs/user-facing-errors/src/schema_engine.rs (hunk 2 above)
Comment (Druue) [reply libs/user-facing-errors/src/schema_engine.rs:270]: Cool so this was intentionally removed as clean-up, good to know :) Need to double check if this still exists in prisma/prisma too

File: schema-engine/sql-migration-tests/tests/initialization/mod.rs (hunk 4 above)
Comment (Druue) [reply schema-engine/sql-migration-tests/tests/initialization/mod.rs:56]: cool, no worries

**Review by tomhoule — APPROVED** [review]
